gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDataset.java

```diff
@@ -188,11 +215,19 @@ protected Map<Path, FileStatus> getFilePathsToFileStatus(FileSystem targetFs, Co
           this.getFileSetId(), currentSnapshotOverview.getSnapshotId(),
           currentSnapshotOverview.getManifestListPath(),
           currentSnapshotOverview.getMetadataPath().orElse("<<ERROR: MISSING!>>"));
-      return Maps.newHashMap();
+      TableMetadata readTimeTableMetadata = currentSnapshotOverview.getTableMetadata().orElseThrow(() -> new RuntimeException(
+          String.format("~%s~ no table metadata for current snapshot '%s' at '%s' with metadata path '%s'",
+              this.getFileSetId(), currentSnapshotOverview.getSnapshotId(),
+              currentSnapshotOverview.getManifestListPath(),
+              currentSnapshotOverview.getMetadataPath().orElse("<<ERROR: MISSING!>>"))));
+      return new GetFilePathsToFileStatusResult(Maps.newHashMap(), readTimeTableMetadata);
     }
+
+    List<TableMetadata> readTimeTableMetadataHolder = Lists.newArrayList(); // expecting exactly one elem
```

Comment:
Why not just `TableMetadata readTimeTableMetadataHolder = null` if there will be only one element?

Reply:
You can filter out absents and count the size -f filePathsIterator in place of doing `eadTimeTableMetadataHolder.size()` later

Reply:
I really work to avoid null, because nothing about it indicates the need to pre-check before use.  (the way we all know to check `List::size` before `List::get`).

but the bigger issue is that a java Consumer isn't allowed to reassign, so this is not legal:
```
Optional<X> otherOptional = ... 
Optional<X> myVar = Optional.empty();
otherOptional.ifPresent(val -> {
  myVar = Optional.of(val)
});
```
since `myVar` must be "effectively final" to be used in a java closure

Reply:
you do bring up a valid point about exactly one element, so I added a check for that.  in the end it felt easier and slightly more correct to have any num MD != 1 be a fatal error (than a warning)